tests/signature_pad.test.ts

```diff
@@ -119,22 +119,116 @@ describe('user interactions', () => {
         clientX: 50,
         clientY: 30,
         pressure: 1,
-      })
+      }),
     );
     canvas.dispatchEvent(
       new PointerEvent('pointerdown', {
         clientX: 240,
         clientY: 30,
         pressure: 1,
-      })
+      }),
     );
     canvas.dispatchEvent(
       new PointerEvent('pointerdown', {
         clientX: 150,
         clientY: 120,
         pressure: 1,
-      })
+      }),
     );
     expect(pad.toDataURL('image/svg+xml')).toMatchSnapshot();
   });
 });
+
+describe('Signature events.', () => {
+  let signpad: SignaturePad;
+  let eventDispatched: Event;
+
+  const eventHandler: EventListener = (evt: Event): void => {
+    eventDispatched = evt;
+  };
+
+  beforeEach(() => {
+    signpad = new SignaturePad(canvas);
+
+    // to make this test works, canvas must be added to the document body.
+    document.body.insertAdjacentElement('afterbegin', canvas);
+  });
+
+  afterEach(() => {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    eventDispatched = undefined!;
+
+    document.body.removeChild(canvas);
+  });
+
+  [
+    { eventName: 'beginStroke', dispatchedEventName: 'pointerdown' },
+    { eventName: 'beforeUpdateStroke', dispatchedEventName: 'pointerdown' },
+    { eventName: 'afterUpdateStroke', dispatchedEventName: 'pointerdown' },
+    { eventName: 'endStroke', dispatchedEventName: 'pointerup' },
+  ].forEach((param) => {
+    describe(`${param.eventName}.`, () => {
+      beforeEach(() => {
+        signpad.addEventListener(param.eventName, eventHandler);
+      });
+
+      afterEach(() => {
+        signpad.removeEventListener(param.eventName, eventHandler);
+      });
+
+      it('no writing to the canvas.', () => {
+        expect(eventDispatched).toBeFalsy();
+      });
+
+      it('writes to the canvas.', () => {
+        const eventInitObj = <PointerEventInit>{
+          clientX: 50,
+          clientY: 30,
+          pressure: 1,
+          bubbles: true,
+        };
+        const pointerEvent = new PointerEvent(
+          param.dispatchedEventName,
+          eventInitObj,
+        );
+        canvas.dispatchEvent(pointerEvent);
+
+        expect(eventDispatched).toBeTruthy();
+        expect(eventDispatched).toBeInstanceOf(CustomEvent);
+
+        const event = <CustomEvent>eventDispatched;
+        expect(event.detail).toBe(pointerEvent);
+      });
+    });
+  });
+
+  [{ useDocument: true }, { useDocument: false }].forEach((param) => {
+    describe(`use document as EventTarget=${param.useDocument}.`, () => {
+      beforeEach(() => {
+        signpad['_et'] = document;
```

Comment:
```suggestion
        if (param.useDocument) {
          signpad['_et'] = document;
        }
```

Reply:
ah thanks for reminding me about this. But I think I will implement it differently since testing with EventTarget object has been done at previous test case (above). I will provide with a new commit.